pennylane_lightning/lightning_tensor/backends/quimb/_mps.py

```diff
@@ -79,22 +214,215 @@ def __init__(self, num_wires, interf_opts, dtype=np.complex128):
+
+    def _apply_operation(self, op: qml.operation.Operator) -> None:
+        """Apply a single operator to the circuit, keeping the state always in a MPS form.
+
+        Internally it uses `quimb`'s `apply_gate` method. For operations that act on more than two wires,
+        it decomposes them first into operations that act on at most two wires.
+
+        Args:
+            op (Operator): The operation to apply.
+        """
+
+        if len(op.wires) <= 2:
+            self._circuitMPS.apply_gate(op.matrix(), *op.wires, **self._gate_opts)
+        else:
+            decom_ops = decompose_recursive(op)
+            for o in decom_ops:
+                self._circuitMPS.apply_gate(o.matrix(), *o.wires, **self._gate_opts)
+
+    def measurement(self, measurementprocess: MeasurementProcess) -> TensorLike:
+        """Measure the measurement required by the circuit over the MPS.
+
+        Args:
+            measurementprocess (MeasurementProcess): measurement to apply to the state.
+
+        Returns:
+            TensorLike: the result of the measurement.
+        """
+
+        return self._get_measurement_function(measurementprocess)(measurementprocess)
+
+    def _get_measurement_function(
+        self, measurementprocess: MeasurementProcess
+    ) -> Callable[[MeasurementProcess, TensorLike], TensorLike]:
+        """Get the appropriate method for performing a measurement.
+
+        Args:
+            measurementprocess (MeasurementProcess): measurement process to apply to the state
+
+        Returns:
+            Callable: function that returns the measurement result
+        """
+        if isinstance(measurementprocess, StateMeasurement):
+            if isinstance(measurementprocess, ExpectationMP):
+                return self.expval
+
+            if isinstance(measurementprocess, VarianceMP):
+                return self.var
+
+        raise NotImplementedError
+
+    def expval(self, measurementprocess: MeasurementProcess) -> float:
+        """Expectation value of the supplied observable contained in the MeasurementProcess.
+
+        Args:
+            measurementprocess (StateMeasurement): measurement to apply to the MPS.
+
+        Returns:
+            Expectation value of the observable.
+        """
+
+        obs = measurementprocess.obs
+
+        result = self._local_expectation(obs.matrix(), tuple(obs.wires))
+
+        return result
+
+    def var(self, measurementprocess: MeasurementProcess) -> float:
+        """Variance of the supplied observable contained in the MeasurementProcess.
+
+        Args:
+            measurementprocess (StateMeasurement): measurement to apply to the MPS.
+
+        Returns:
+            Variance of the observable.
+        """
+
+        obs = measurementprocess.obs
+
+        obs_mat = obs.matrix()
+        expect_squar_op = self._local_expectation(np.dot(obs_mat, obs_mat), tuple(obs.wires))
+        expect_op = self._local_expectation(obs_mat, tuple(obs.wires))
```

Comment:
Could we call `expval`? It would be better for maintainability I reckon.

Reply:
Maybe I'm missing something, but introducing the `_local_expectation` method was precisely for maintainability. The idea is that `expval` and `var` operate just by calling `_local_expectation` (which uses `quimb` internally). Therefore, if we modify the latter, we operate on the `expval` and `var` methods at the same time. Furthermore, `expval` receives a `MeasurementProcess` argument, which is not what we want to pass it here. What do you think?

Reply:
A bit of a nitpick since `expval` is almost a one-liner. But `self._local_expectation(obs_mat, tuple(obs.wires))` is the core implementation of `expval`, so suppose there is a bug with this line, or the private method `_local_expectation` disappears, then you would have to fix the bug in two places instead of one. It is in that sense that it is more maintainable.

Reply:
I implemented the change, although I wouldn't say I like that we call the `compute_matrix` method twice this way. Hopefully, this doesn't affect performance significantly